src/metabase/api/card.clj

```diff
@@ -125,6 +128,20 @@
   [_filter-option model-id]
   (cards-for-segment-or-metric :segment model-id))
 
+(defmethod cards-for-filter-option* :stale
+  [_]
+  (let [card-id->field-usages (group-by :card_id (t2/select :model/FieldUsage
+                                                            {:select [:fu.* [:f.name :new_column_name] [:t.name :new_table_name]]
+                                                             :from   [[(t2/table-name :model/FieldUsage) :fu]]
+                                                             :join   [[(t2/table-name :model/Field) :f] [:= :f.id :fu.field_id]
+                                                                      [(t2/table-name :model/Table) :t] [:= :t.id :f.table_id]]
+                                                             :where  [:= false :fu.is_current]}))
+        card-ids              (keys card-id->field-usages)]
+    (if (empty? card-ids)
+      []
+      (map #(assoc % :field_usages (get card-id->field-usages (:id %)))
+           (t2/select Card :id [:in card-ids] default-order)))))
```

Comment:
I can be a clown about more complex hydration cases like this. Doing it without toucan's `hydrate` seems as efficient as it could be and like fairly clear code, but if there's a case for using `hydrate` I'm happy to hear it.

Reply:
I'd love someone to take me through hydration works, and until then I don't have a strong opinion.

I would however hoist up the append function, big readability improvement IMHO.

I think it would even be nice to just make it a top-level function that gets curried with the map:

```clojure
(defn- assoc-field-usages [card-id->field-usages {:keys [id] :as card}]
  (assoc card :field_usages (card-id->field-usages id)))
```

I know this level of decomposition is not for everyone though.